middle_end/flambda2/terms/function_declarations.ml

```diff
@@ -43,8 +43,10 @@ let [@ocamlformat "disable"] print ppf { in_order; _ } =
 
 let free_names { funs; _ } =
   Closure_id.Map.fold
-    (fun _closure_id code_id syms ->
-      Name_occurrences.add_code_id syms code_id Name_mode.normal)
+    (fun closure_id code_id syms ->
+      Name_occurrences.add_closure_id
+        (Name_occurrences.add_code_id syms code_id Name_mode.normal)
+        closure_id Name_mode.normal)
```

Comment:
This should probably also be reverted: since we do not count the var_within_closure bound by a set of closures, we shouldn't count the closure_ids bound either.

Reply:
done